packages/icon/src/createIconComponent.tsx

```diff
@@ -14,10 +14,15 @@ type GlyphObject = Record<string, LGGlyph.Component>;
 export default function createIconComponent<
   G extends GlyphObject = GlyphObject
 >(glyphs: G) {
-  const Icon = ({ glyph, ...rest }: IconProps) => {
+  const Icon = ({ glyph, size, ...rest }: IconProps) => {
     const SVGComponent = glyphs[glyph];
+    let iconSize = sizeMap[Size.Default];
 
-    return <SVGComponent {...rest} />;
+    if (size !== undefined) {
+      iconSize = typeof size === 'number' ? size : sizeMap[size];
+    }
+
+    return <SVGComponent size={iconSize} {...rest} />;
```

Comment:
It doesn't seem like this works unless the consuming developer handles the own `size` prop in the custom svg themselves
(i.e. 
```
const myGlyphs = {
  MyCustomGlyph: ({size}) => <svg height={size} width={size} />,
};
```
)

I'm not sure this adds much value over requiring the svg be properly sized in the first place. 

I think we should come up with a way to get the `size` prop to work without needing handling within the custom.